actions/utils.ts

```diff
@@ -359,3 +365,103 @@ export function isComposableCowCompatible(code: string): boolean {
     return code.includes(sighash.slice(2));
   });
 }
+
+type ParsedError = {
+  errorNameOrSelector?: string;
+  message?: string;
+}
+
+/**
+ * Given a raw ABI-encoded custom error returned from a revert, extract the selector and optionally a message.
+ * @param abi of the custom error, which may or may not be parameterised.
+ * @returns an empty parsed error if assumptions don't hold, otherwise the selector and message if applicable.
+ */
+const rawErrorDecode = (abi: string): ParsedError  => {
+  if (abi.length == 10) {
+    return { errorNameOrSelector: abi }
+  } else {
+    try {
+      const selector = String(abi).substring(0, 10);
```

Comment:
`abi` is a string already, I don't think you need to wrap it in `String`

Also, just a note on the method, I've seem mainly `slice` used for this sort of thing, while I don't think I ever saw `substring`.

Comparison: https://stackoverflow.com/a/31910656/1272513

Not that it matters

Reply:
Nice, thanks for the tips :+1: